### ISSUE TITLE
Why C in Log Regression is inverse of regularization strength?

### ISSUE THREAD

**Issue by Kirushikesh** [ageron/handson-ml2#524]
Hi @ageron ,

When discussing about Logistic Regression, you said that C denotes the inverse of regularization strength, that is the lower value of C generalizes well right? A similar type of notation is used in SVM for soft margin where I understood it how. Can you please give more clarification on how it works on Logistic Regression??

Also LogisticRegression() uses C inversely proportional to regularization but SGDClassifier(loss="log") uses alpha that directly proportional to regularization. But both models are the same right?

**Comment by Kirushikesh** [comment]
I still have this question in my mind does anyone knows the answer?

**Comment by ageron** [comment]
Hi @Kirushikesh ,
The lower C is the *more* the model is regularized. It doesn't mean it will necessarily generalize better, you always need to find the optimal level of regularization (e.g., using cross-validation).
The discussion in this issue is useful: https://github.com/scikit-learn/scikit-learn/issues/21891
Hope this helps!

**Comment by Kirushikesh** [comment]
How smaller the value of C more the model is regularized? Can you explain this statement in detail

**Comment by ageron** [comment]
Hi @Kirushikesh ,
The mathematical equations for most Scikit-Learn models are available in their User Guide. For `LogisticRegression`, it's [here](https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression):
<img width="897" alt="image" src="https://user-images.githubusercontent.com/76661/155825332-1b8c3372-1561-4bc1-814b-72f8e76d4687.png">

Notice that the first term is the regularization term (with 3 variants: l1, l2, and a mix of both), and the second term is the log loss. Importantly, note that the hyperparameter C multiplies the *second* term, not the regularization loss. But of course, minimizing the sum **reg + C*log_loss** is equivalent to minimizing the loss **1/C*reg + log_loss**. So C is literally the inverse of the regularization strength. If you multiply C by 10, you divide the regularization term by 10.

Hope this helps.

**Comment by ageron** [comment]
Great, I'm glad this helped.
I'm not sure why the researchers used `alpha * main_loss + regularization_loss` in one case and `main_loss + C * regularization_loss` in other cases, it's probably just historical.

**Comment by Kirushikesh** [comment]
Thanks sir that helps
